Automattic/jetpack
# Social: Update enhanced publishing feature check

## Proposed changes:

~This builds on #29723 which introduces the
`has_enhanced_publishing_feature` method, which is a more efficient feature check, using `Current_Plan::supports`, relying on the heartbeat mechanism to update the local plan information rather than calling the API endpoint.~

As #29723 is now abandoned this is updated to bring the `has_enhanced_publishing_feature` method over, which this change relies upon.

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
For both Social and Jetpack:
* Make sure the site is connected with the auto-sharing enabled.
* Load the editor, and check the sidebar
* The 'Media' section should not be shown.
* Buy the Advanced plan
* Load the editor again, and the 'Media' section will be there

<img width="258" alt="image" src="https://user-images.githubusercontent.com/96462/229581359-a58f00cc-0649-479a-91d6-20f08f0747c2.png">
